lsp/ruby_lsp.lua

```diff
@@ -20,8 +20,12 @@ return {
   init_options = {
     formatter = 'auto',
   },
+  on_new_config = function(config, root_dir)
+    config.cmd_cwd = root_dir
+  end,
   reuse_client = function(client, config)
-    config.cmd_cwd = config.root_dir
-    return client.config.cmd_cwd == config.cmd_cwd
+    local client_cwd = client.config.cmd_cwd or client.root_dir
+    local config_cwd = config.cmd_cwd or config.root_dir
+    return client_cwd == config_cwd
```

Comment:
why is this thing forcing root_dir to always be equal to CWD? 

i would think this whole `reuse_client` can be dropped, assuming the other comment is addressed.